speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/ResearcherMenu.cpp

```diff
@@ -1,75 +1,259 @@
+/// @brief Handle input in the UserID textbox
+static void SetUserID(void*)
+{
+    char* m_userID = GfuiEditboxGetString(s_scrHandle, m_userIDId);
+    GfuiEditboxSetString(s_scrHandle, m_userIDId, m_userID);
 }
 
 
-/// @brief Initializes the researcher menu
-/// @param nextMenu The scrHandle of the next menu
-/// @return The researcherMenu scrHandle
+
+/// @brief Saves the settings into the frontend settings and the backend config
+static void SaveSettings(void* /* dummy */)
+{
+    // Save settings to backend config
+    SMediator* mediator = SMediator::GetInstance();
+    mediator->SetTask(m_task);
+    mediator->SetIndicatorSettings(m_indicators);
+    mediator->SetInterventionType(m_interventionType);
+    mediator->SetMaxTime(m_maxTime);
+    mediator->SetUserID(m_userID);
+
+    // Save settings to frontend settings
+    // TODO: Set Environment (Track)
+    // TODO: Set Participant control (bool*)
+
+    // Go to the next screen
+    GfuiScreenActivate(s_nextHandle);
+}
+
+/// @brief            Initializes the researcher menu
+/// @param p_nextMenu The scrHandle of the next menu
+/// @return           The researcherMenu scrHandle
 void* ResearcherMenuInit(void* p_nextMenu)
 {
     // Return if screen already created
     if (s_scrHandle) {
         return s_scrHandle;
     }
-
     // Otherwise, create the screen
     s_scrHandle = GfuiScreenCreate((float*)NULL, NULL, OnActivate,
                                    NULL, (tfuiCallback)NULL, 1);
     s_nextHandle = p_nextMenu;
 
-    void *param = GfuiMenuLoad("ResearcherMenu.xml");
+    void* param = GfuiMenuLoad("ResearcherMenu.xml");
     GfuiMenuCreateStaticControls(s_scrHandle, param);
 
-    // InterventionType controls: arrow buttons, label hover-ability
-    GfuiMenuCreateButtonControl(s_scrHandle,param,"InterventionLeftArrow",(void*)-1,ChangeInterventionType);
-    GfuiMenuCreateButtonControl(s_scrHandle,param,"InterventionRightArrow",(void*)1,ChangeInterventionType);
-    s_interventionTypeId = GfuiMenuCreateLabelControl(s_scrHandle,param,"InterventionLabel");
+    // Task checkboxes controls
+    GfuiMenuCreateCheckboxControl(s_scrHandle, param, "CheckboxTask1", NULL, SelectLaneKeeping);
+    GfuiMenuCreateCheckboxControl(s_scrHandle, param, "CheckboxTask2", NULL, SelectSpeedControl);
+
+    // Indicator checkboxes controls
+    GfuiMenuCreateCheckboxControl(s_scrHandle, param, "CheckboxIndicator1", NULL, SelectAuditory);
+    GfuiMenuCreateCheckboxControl(s_scrHandle, param, "CheckboxIndicator2", NULL, SelectVisual);
+
+    // Types checkboxes controls
+    GfuiMenuCreateCheckboxControl(s_scrHandle, param, "CheckboxType1", NULL, SelectTypeNoSignals);
+    GfuiMenuCreateCheckboxControl(s_scrHandle, param, "CheckboxType2", NULL, SelectTypeOnlySignals);
+    GfuiMenuCreateCheckboxControl(s_scrHandle, param, "CheckboxType3", NULL, SelectTypeSharedControl);
+    GfuiMenuCreateCheckboxControl(s_scrHandle, param, "CheckboxType4", NULL, SelectTypeCompleteTakeover);
```

Comment:
CheckboxType1 is quite a generic name, maybe something more specific like CheckboxTypeNoSignals would be better